src/wallet/walletdb.h

```diff
@@ -314,6 +314,61 @@ class ZcashdUnifiedAddressMetadata {
     }
 };
 
+// Serialization wrapper for reading and writing RecipientAddress
+// in CompactSize format.
+class CSerializeRecipientAddress {
+    libzcash::ReceiverType typecode;
+
+    public:
+        libzcash::RecipientAddress recipient;
+        CSerializeRecipientAddress() {} // for serialization only
```

Comment:
The way that I've dealt with this elsewhere is that I make this constructor private, then provide an explicit `Typename::Read` method that takes a stream as argument. You can see how this is done in `SaplingDiversifiableFullViewingKey::Read`, for example.

Reply:
I copied the public constructor pattern from `ReceiverTypeSer`.

When I tried the `Typename::Read` method, I got stuck on having to implement `Serialize` for `RecipientAddress`, which is a variant.

Reply:
So, you could do it here very simply:
```
template <typename Stream>
static libzcash::RecipientAddress CSerializeRecipientAddress::Read(Stream& stream) {
  CSerializeRecipientAddress csr;
  stream >> csr;
  return csr.recipient;
}
```